app/uk/gov/hmrc/transitmovements/controllers/MovementsController.scala

```diff
@@ -136,7 +129,37 @@ class MovementsController @Inject() (
       )
   }
 
-  def updateMovement(movementId: MovementId, triggerId: Option[MessageId] = None): Action[Source[ByteString, _]] =
+  def updateMovement(movementId: MovementId, triggerId: Option[MessageId] = None) =
+    contentTypeRoute {
+      case Some(_) => updateMovementSmallMessage(movementId, triggerId)
+      case None    => updateMovementLargeMessage(movementId, triggerId)
+    }
+
+  def updateMovementLargeMessage(movementId: MovementId, triggerId: Option[MessageId] = None): Action[AnyContent] = Action.async(parse.anyContent) {
+    implicit request =>
+      (for {
+        messageType    <- extract(request.headers).asPresentation
+        objectStoreURL <- extractObjectStoreURI(request.headers).asPresentation
+        objectStoreFilePath = if (objectStoreURL.contains("/")) objectStoreURL.split("/", 2).apply(1) else objectStoreURL
```

Comment:
can we do this in the objectStore object and returned as a value instead?

Reply:
I agree, it's the service's job to do this, not the presentation layer's.